enterprise/views.py

```diff
@@ -886,74 +973,18 @@ def post(self, request):
-                    )
-                    course_mode = 'verified' if 'verified' in course_modes \
-                        else 'professional' if 'professional' in course_modes \
-                        else 'no-id-professional' if 'no-id-professional' in course_modes \
-                        else 'audit'
-                    try:
-                        enrollment_api_client.enroll_user_in_course(
-                            request.user.username,
-                            course_id,
-                            course_mode
-                        )
-                        LOGGER.info(
-                            'Created LMS enrollment for User {user} in Course {course_id} '
-                            'with License {license_uuid} in Course Mode {course_mode}.'.format(
-                                user=request.user.username,
-                                course_id=course_id,
-                                license_uuid=license_uuid,
-                                course_mode=course_mode
-                            )
-                        )
-                    except Exception as exc:    # pylint: disable=broad-except
-                        LOGGER.error(
-                            'Unable to create an LMS enrollment from the DSC view: {exc}'.format(
-                                exc=exc
-                            )
-                        )
-                        return redirect(failure_url)
-                try:
-                    self.create_enterprise_course_enrollment(request, consent_record, course_id, license_uuid)
-                except IntegrityError:
-                    error_code = 'ENTGDS009'
-                    log_message = (
-                        '[Enterprise DSC API] IntegrityError while creating EnterpriseCourseEnrollment.'
-                        'Course: {course_id}, '
-                        'Program: {program_uuid}, '
-                        'EnterpriseCustomer: {enterprise_customer_uuid}, '
-                        'User: {user_id}, '
-                        'License UUID: {license_uuid}, '
-                        'ErrorCode: {error_code}'.format(
-                            course_id=course_id,
-                            program_uuid=program_uuid,
-                            enterprise_customer_uuid=enterprise_uuid,
-                            user_id=request.user.id,
-                            license_uuid=license_uuid,
-                            error_code=error_code,
-                        )
-                    )
-                    LOGGER.exception(log_message)
+            successful_enrollment = self._enroll_leaner_in_course(
+                request=request,
+                enterprise_customer=enterprise_customer,
+                course_id=course_id,
+                program_uuid=program_uuid,
+                license_uuid=license_uuid)
             consent_record.granted = consent_provided
             consent_record.save()
```

Comment:
should the consent record still be updated if the enrollment fails?

Reply:
It looks like it was originally. I think we should maintain the logic how it was for now and investigate whether or not the consent record should be updated later.

Reply:
previously, the redirect (on failure) would happen before the consent record is updated so to keep with existing logic it shouldn't be updated after failure here

Reply:
another reason to wrap these calls in a try/except and rethrow exceptions from within the enrollment util method if you ask me 🤷‍♂️